src/null-safety/migrating-to-null-safety/index.md

```diff
@@ -0,0 +1,435 @@
+    some Dart code, *then* discover the language version marker and then
+    potentially have to re-parse code it already parsed.
+
+*   The comment applies to an entire *library*. If the library uses part files,
+    they also *all* must have identical version comments. We require the
+    redundant markers in all part files so that Dart tools that work on a
+    per-file basis can still unambiguously determine what version a part file is
+    at without having to find the corresponding library file.
+
+*   The marker should only move the library to a *lower* language version than
+    the default version specified by the pubspec. If you could opt a library
+    into a version *higher* than the pubspec allows, you could get into a
+    situation where a user of an older Dart SDK would attempt to use your
+    package even though it contains libraries whose language version isn't
+    supported by their SDK.
+
+    For example, say your package's SDK constraint is `>=2.6.0 <3.0.0`, giving
+    you a default language version of 2.6. Someone using Dart SDK 2.6.0 might be
+    using your package. If you have a marker like `// @dart=2.5`, everything is
+    fine because Dart SDK 2.6.0 also supports language 2.5. But if you were to
+    try to put `// @dart=2.7` there, you would end up with a library whose
+    version isn't supported by the 2.6.0 SDK the use has.
+
+    In terms of null safety, this rule means you'll use the pubspec SDK
+    constraint to opt your package *in* to null safety and can then use comment
+    markers to opt individual libraries *out* if needed.
+
+### New code and the default default language version
+
+Dart doesn't require you to put all of your code inside a pub package, though
+most Dart code is. Pub also doesn't require your pubspec to have an SDK
+constraint. In either of those cases, what language version do you get?
+
+Since our goal is a world where all Dart code is modern and null safe, you get
+the latest language version in this case. That means if you sit down with a text
+editor, make a brand new `.dart` file, and run it, you get null safety. Add a
+pubspec and you're still using null safety. Unless you write an SDK constraint
+that explicitly opts you *out* of null safety, we keep you in the new world.
+
+When you start depending on other packages, pub keeps you in that new world. If
+your package is using null safety, then pub tries to only give you packages that
+also use null safety.
+
+**TODO: Is the above correct?**
+
+## Mixed-mode programs
+
+What if your null safe app depends on a package that isn't null safe? More
+generally, what happens when your program contains some libraries at a null safe
+language version and others that are using a legacy language? Is this allowed?
+
+The answer is "yes". We called these **mixed-mode programs.** ("Mixed-version"
+would probably be more accurate, but the name has kind of stuck.) In the context
```

Comment:
"has kind of stuck": have we used the term externally? If not, I guess it hasn't really stuck there, and we can just use what we think is the better term...

Reply:
I personally prefer mixed-mode, that's why I use it, but... *shrug*.

Reply:
I have no preference for the actual term, but would strongly prefer that we settle on a single term

Reply:
I like mixed-mode too. Removed the parenthetical to make it sound more official.